dotnet/src/SemanticKernel/Reliability/RetryWithDelay.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.Diagnostics;
+using static Microsoft.SemanticKernel.Configuration.KernelConfig;
+
+namespace Microsoft.SemanticKernel.Reliability;
+
+/// <summary>
+/// A retry mechanism that retries once after a delay, using RetryAfter header if available.
+/// </summary>
+public class RetryWithDelay : IRetryMechanism
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RetryWithDelay"/> class.
+    /// </summary>
+    /// <param name="config">The retry configuration.</param>
+    public RetryWithDelay(RetryConfig? config = null) : this(config!, null, null)
+    {
+        this._config = config ?? new RetryConfig();
+    }
+
+    internal RetryWithDelay(RetryConfig config, IDelayProvider? delayProvider = null, ITimeProvider? timeProvider = null)
```

Comment:
I might have gone overboard here, but really wanted UTs here to have good coverage and be reliable. Thoughts @dluc ?

Reply:
Resolving for now as-is. Create new conversation as appropriate.